pubsub/pulsar/pulsar.go

```diff
@@ -123,16 +125,29 @@ func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
 		producer = cache.(pulsar.Producer)
 	}
 
-	_, err = producer.Send(context.Background(), &pulsar.ProducerMessage{
-		Payload: req.Data,
-	})
-	if err != nil {
+	msg := parsePublishMetadata(req)
+	if _, err = producer.Send(context.Background(), msg); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// parsePublishMetadata parse publish metadata
+func parsePublishMetadata(req *pubsub.PublishRequest) (msg *pulsar.ProducerMessage) {
+	msg = &pulsar.ProducerMessage{
+		Payload: req.Data,
+	}
+	if val, ok := req.Metadata[deliverAt]; ok {
+		msg.DeliverAt, _ = time.Parse("2006-01-02 15:04:05", val)
```

Comment:
The https://github.com/dapr/components-contrib/pull/1107#discussion_r699544119 still applies:

- Is there a reason to use a custom time format instead of `time.RFC3339`? 
- Do not ignore the errors returned from `time.Parse` and `time.ParseDuration`. The function signature should be updated to also return the error, which can be checked and returned by the calling `Publish()` function.

```go
func parsePublishMetadata(req *pubsub.PublishRequest) (*pulsar.ProducerMessage, error) 
```

Reply:
updated.